app/components/Layout.tsx

```diff
@@ -0,0 +1,235 @@
+        <Dialog as="div" className="relative z-40 lg:hidden" onClose={setOpen}>
+          <Transition.Child
+            as={Fragment}
+            enter="transition-opacity ease-linear duration-300"
+            enterFrom="opacity-0"
+            enterTo="opacity-100"
+            leave="transition-opacity ease-linear duration-300"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <div className="fixed inset-0 bg-black bg-opacity-25" />
+          </Transition.Child>
+
+          <div className="fixed inset-0 z-40 flex">
+            <Transition.Child
+              as={Fragment}
+              enter="transition ease-in-out duration-300 transform"
+              enterFrom="-translate-x-full"
+              enterTo="translate-x-0"
+              leave="transition ease-in-out duration-300 transform"
+              leaveFrom="translate-x-0"
+              leaveTo="-translate-x-full"
+            >
+              <Dialog.Panel className="relative flex w-full max-w-xs flex-col overflow-y-auto bg-white pb-12 shadow-xl">
+                <div className="flex px-4 pt-5 pb-2">
+                  <button
+                    type="button"
+                    className="-m-2 inline-flex items-center justify-center rounded-md p-2 text-gray-400"
+                    onClick={() => setOpen(false)}
+                  >
+                    <span className="sr-only">Close menu</span>
+                    <XMarkIcon className="h-6 w-6" aria-hidden="true" />
+                  </button>
+                </div>
+                {/* Links */}
+                <div className="space-y-6 border-t border-gray-200 py-6 px-4">
+                  {collections?.map((collection: Collection) => (
+                    <div key={collection.node.title} className="flow-root">
+                      <a
+                        href={"/"}
+                        className="-m-2 block p-2 font-medium text-gray-900"
+                      >
+                        {collection.node.title}
+                      </a>
+                    </div>
+                  ))}
+                </div>
+              </Dialog.Panel>
+            </Transition.Child>
+          </div>
+        </Dialog>
+      </Transition.Root>
+    );
+  }
+
+  function Cart() {
+    return (
+      <div className="ml-4 flow-root lg:ml-8">
+        <a
+          href={cartId ? `/cart/${cartId}` : "/"}
+          className="group -m-2 flex items-center p-2"
+        >
+          <ShoppingBagIcon
+            className="h-6 w-6 flex-shrink-0 text-gray-400 group-hover:text-gray-500"
+            aria-hidden="true"
+          />
+          <span className="ml-2 text-sm font-medium text-gray-700 group-hover:text-gray-800">
+            0
+          </span>
+          <span className="sr-only">items in cart, view bag</span>
+        </a>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <header className="relative">
```

Comment:
I'm confused, is this the header or the cart component? seems like the header, so name it like that

Reply:
Committed new Layout structure